qcodes/instrument_drivers/rigol/DS4000.py

```diff
@@ -250,3 +252,11 @@ def __init__(self, name, address, timeout=20, **kwargs):
 
         channels.lock()
         self.add_submodule('channels', channels)
+
+    def _check_firmware_version(self):
+        #Require version 00.02.03
+
+        idn = self.get_idn()
+        ver = [int(x) for x in idn['firmware'].split('.')]
+        if ver < [0,2,3]:
+            raise RuntimeError('Firmware version must be at least 00.02.03')
```

Comment:
also, do you receive corrupted data all the time? as in, is the scope still usable to some extent with `00.02.02` and lower or not? if it is, then please change `raise RuntimeError` to a waning `warnings.warn`, because the users should not get suddenly blocked from using the driver/instrument.

Reply:
Also, I just trust you that it is indeed the firmware problem, and not some minor error in the way you are using the instrument. Nevertheless, could you please double check that you are doing things in the right way?

Reply:
Sincerely I don't know how extended is the breakage. I just noticed that the function to get the raw buffer was broken and got working after an upgrade. Also now I don't have a way anymore to check with the old version. Since it's easily upgredable, I would not waste time to try to use an old version

Reply:
ok. but still let's keep it safe. So:
- make a warning instead of error
- add more info to the text of the warning explaining what is wrong with an older firmware